Igire-Rwanda-Systems-Workshop/IRO-website-bn
# Add Real-Time Notification Feature

Description: This PR adds the real-time notification feature for users based on their roles when a payment is recorded. Notifications are sent to both the Operations Manager and the Project Director using WebSocket, allowing them to receive instant updates.

Changes Made:

Created real-time notifications using WebSocket for the Operations Manager and Project Director upon recording a payment.
Integrated the WebSocket server to emit notifications when a payment is successfully recorded.
Created notification messages to be stored in the database with status tracking.
Set up role-based notifications:
Operations Manager receives a notification for each payment update related to their requests.
Project Director receives notifications when a payment is made.
Error handling for WebSocket connections and notification delivery.
How to Test:

Record a payment via the system.
Verify that the Operations Manager and Project Director receive real-time notifications.
Check that the notifications are stored in the database with the correct status (unread or read).
